internal/core/domain/fee.go

```diff
@@ -0,0 +1,17 @@
+package domain
+
+import "context"
+
+type IntentFees struct {
```

Comment:
nit: it makes sense to have it in domain but looks duplicate of `IntentFeeInfo` defined in application/types.go. let's reuse domain.IntentFees there ? wdyt ?